### PR TITLE
DEV: FIX Requirements Versions

### PR DESCRIPTION
**Pull request by shaneahmed**
- Update requirements.
- Sphinx 4.5.0 does not resolve on python 3.7 now. The requirement has been downgraded to 4.3.2 and checked html build locally. It passes the build and the output html looks fine.
- I have changed the base requirements to a single file `requirements.txt`. `requirements_dev.txt` installs any additional dependencies on top of base requirements. conda environment is also based on `requirements.txt` and `requirements_dev.txt` now.
- This removes the need for pre-commit check. However, I think we should modify the pre-commit check according to this update.

### CODE REVIEW

File: .github/workflows/conda-env-create.yml [hunk 1]
```diff
@@ -4,27 +4,42 @@ name: Solve Conda Environment
...
+            kind: prod
             requirements: requirements.win64.conda.yml
+          - os: windows-latest
+            kind: dev
+            requirements: requirements.win64.dev.conda.yml
     runs-on: ${{ matrix.os }}
     timeout-minutes: 20
     steps:
       - uses: actions/checkout@v2
+      - name: Copy file
```
Comment (John-P) [inline .github/workflows/conda-env-create.yml:40]:
```suggestion
      - name: Copy requirements files
```

**Review by John-P — APPROVED** [review]
A couple of minor points but I'm happy that this works.

File: .github/workflows/conda-env-create.yml (hunk 1 above)
Comment (shaneahmed) [reply .github/workflows/conda-env-create.yml]: Addressed in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/455/commits/59f01e6e48eadfc58a736558303f09cbc16276b0

File: .github/workflows/conda-env-create.yml [hunk 2]
```diff
@@ -4,27 +4,42 @@ name: Solve Conda Environment
...
-    paths: [ "requirements*.yml", "conda-env-create.yml" ]
+    paths: [ "requirements*.yml", "conda-env-create.yml", "requirement*.txt"]
   schedule: # Run on the 1st of every month at midnight
     - cron: "0 0 1 * *"
 
 jobs:
   conda_solve:
+    defaults:
+      run:
+        working-directory: .
```
[the remark this answers is not in the document]
Comment (shaneahmed) [reply .github/workflows/conda-env-create.yml]: Addressed in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/455/commits/59f01e6e48eadfc58a736558303f09cbc16276b0

**Review by mostafajahanifar — APPROVED** [review]
Hi Shan,

This looks good. If it works for all the configurations, I'm happy for you to merge it 💯

**Pull request merged**